test/object-store/sync/session/progress_notifications.cpp

```diff
@@ -1333,18 +1327,10 @@ struct EstimatesAreValid : Catch::Matchers::MatcherGenericBase {
         if (size == 1)
             return true;
 
-        // The actual progress for the first message should be the number of
-        // objects downloaded divided by the total number of objects, but in
-        // practice the server starts with a lower estimate so that's only an
-        // upper bound.
-        double expected_first = double(initial_object_count + 1) / (initial_object_count + size);
-        if (entries.front() > expected_first + .01)
-            return false;
-
         // As each of our DOWNLOAD messages have a fixed size, the progress
         // estimate should go up by the same amount each time.
-        double expected_step = (1.0 - entries.front()) / (size - 1);
-        for (size_t i = 1; i < size; ++i) {
+        double expected_step = 1.0 / size;
+        for (size_t i = 0; i < size; ++i) {
```

Comment:
Can we always assume that the first entry in the list is 0? 
What if the progress notification was registered while the transfer has already been started and the first one or more progress items were missed?
(I am thinking of one of my planned test cases for the client reset progress notifications, where the download callback is registered while the fresh realm download is already in progress).

Reply:
The current tests all register the notification before doing anything which would produce notifications. Tests in the future which do something different will need different checks.

Reply:
SGTM - thanks